### PR TITLE
bluestore/NVMEDevice: convert the legacy config opt related with SPDK

### PR DESCRIPTION
**Pull request by optimistyzy**
Change all the legacy config opt related with SPDK in NVMEDevice
 Signed-off-by: Ziye Yang <optimistyzy@gmail.com>

### CODE REVIEW

**Comment by optimistyzy** [comment]
@yuyuyu101 @liupan1111

File: src/os/bluestore/NVMEDevice.cc [hunk 1]
```diff
@@ -365,7 +365,12 @@ void SharedDriverQueueData::_aio_handle(Task *t, IOContext *ioc)
  again:
     dout(40) << __func__ << " polling" << dendl;
     if (current_queue_depth) {
-      spdk_nvme_qpair_process_completions(qpair, g_conf->bluestore_spdk_max_io_completion);
+      r = spdk_nvme_qpair_process_completions(qpair, g_conf->get_val<uint64_t>("bluestore_spdk_max_io_completion"));
```
Comment (yuyuyu101) [inline src/os/bluestore/NVMEDevice.cc]: get_val will be a bit slow than before.  so the better idea is assign to a variable firstly, then uses the variable

File: src/os/bluestore/NVMEDevice.cc [hunk 2]
```diff
@@ -358,14 +358,21 @@ void SharedDriverQueueData::_aio_handle(Task *t, IOContext *ioc)
...
   while (ioc->num_running) {
  again:
     dout(40) << __func__ << " polling" << dendl;
     if (current_queue_depth) {
-      spdk_nvme_qpair_process_completions(qpair, g_conf->bluestore_spdk_max_io_completion);
+      r = spdk_nvme_qpair_process_completions(qpair, max_io_completion);
+      if (r < 0) {
+        ceph_abort();
+      } else if (r == 0) {
+        usleep(io_sleep_in_us);
```
Comment (liupan1111) [inline src/os/bluestore/NVMEDevice.cc]: do we need to sleep here? For the io pattern which needs strict latency request, sleep may cause  latency overhead?

**Comment by tchaikov** [comment]
@liupan1111 please add a Reviewed-by line when merging PRs, also i'd suggest at least push this change to shaman before merging it. so we can be sure it builds.

**Pull request merged**